### DISCUSSION TITLE
[Question] How to resize font

### DISCUSSION THREAD

**Discussion by TornaxO7** [neovide/neovide#2301] [Q&A] [answered]
As the title says, I'm wondering how I can resize the font dynamically in neovide. I can use `Ctrl` + `+` in order to increase the font size in my terminal. Is that also possible in neovide?

**Comment by TornaxO7** [comment]
(Sorry, I accidently submitted the issue way to early here, I'll fix that quickly and reopen it)

**Comment by feature-not-a-bug** [comment]
There's no built-in for changing the size up or down. The simplest option would be to _**set guifont**_ but that can be less than ideal if you're changing the font frequently. With some vimscript you should be able to create a key binding for it.

https://vim.fandom.com/wiki/Change_font_size_quickly

**Comment by MultisampledNight** [comment]
In #856 the OP had this snippet for the resize function itself:

```vim
let s:guifontsize = 10
let s:guifont = "JetBrainsMono\\ Nerd\\ Font"

function! AdjustFontSize(amount)
  let s:guifontsize = s:guifontsize+a:amount
  exe "set guifont=" .. s:guifont .. ":h" .. s:guifontsize
endfunction
```

Is that enough for your purposes? You could map that function to `<C-+>` or `<C-minus>` I guess.

**Comment by TornaxO7** [comment]
Hm... apparently my font setting doesn't get loaded if I have this line:
```vim
set guifont="Fira\\ Code\\ SemiBold:h13"
```
it only works if I have it like this:
```vim
set guifont=Fira\ Code\ SemiBold:h13
```

**Comment by MultisampledNight** [comment]
Fair enough, I referred to the function. My fault for pasting the whole snippet instead.

**Comment by TornaxO7** [comment]
> Fair enough, I referred to the function.

Oh, well since I can't use a string it doesn't look like that I can use this function.

**Comment by MultisampledNight** [comment]
You can use a string. The `\` in there are only for escaping purposes. Your working example will work just fine (if you strip the `:h13` part), the variable is only about having the option locally available.

**Comment by MultisampledNight** [comment]
(offloaded the discussion to Discord, closing)

**Comment by TornaxO7** [comment]
Alright, I have the following settings:

```vim
let s:guifontsize=10
let s:guifont="Fira\\ Code\\ SemiBold"
set guifont=Fira\ Code\ SemiBold:h13

let g:neovide_refresh_rate=60
let g:neovide_transparency=0.8
let g:neovide_no_idle=v:false
let g:neovide_fullscreen=v:false
let g:neovide_cursor_animation_length=0.20
let g:neovide_cursor_trail_length=0.5
let g:neovide_cursor_antialiasing=v:true
let g:neovide_cursor_vfx_mode = "ripple"
let g:neovide_cursor_vfx_opacity=200.0
let g:neovide_cursor_vfx_particle_lifetime=1.2
let g:neovide_cursor_vfx_particle_density=7.0
let g:neovide_cursor_vfx_particle_speed=10.0
let g:neovide_cursor_vfx_particle_phase=1.5
let g:neovide_cursor_vfx_particle_curl=1.0

function! AdjustFontSize(amount)
    let s:guifontsize = s:guifontsize + a:amount
    execute "set guifont=" .. s:guifont .. ":h" .. s:guifontsize
endfunction

nnoremap <C-+> <cmd>call AdjustFontSize(+1)<cr>
nnoremap <C--> <cmd>call AdjustFontSize(-1)<cr>
```
thanks!

**Comment by distek** [comment]
Not sure if this was a thing at the time of this issue, but this is how I'm doing it now:
```lua
local map = vim.keymap.set

local function neovideScale(amount)
	local temp = vim.g.neovide_scale_factor + amount

	if temp < 0.5 then
		return
	end

	vim.g.neovide_scale_factor = temp
end

map("n", "<C-=>", function()
	neovideScale(0.1)
end)

map("n", "<C-->", function()
	neovideScale(-0.1)
end)
```

**Reply by palsmo** [reply]
This solution on my system (Fedora Linux 39) does not refresh the screen, only after another keypress.

**Comment by palsmo** [comment, marked as answer]
This works flawlessly and is the simplest solution, hope it helps!
( *updates the screen immediately, divide by 0 is handled by neovide and just closes the app )
```lua
if vim.g.neovide then
    vim.keymap.set({ "n", "v" }, "<C-+>", ":lua vim.g.neovide_scale_factor = vim.g.neovide_scale_factor + 0.1<CR>")
    vim.keymap.set({ "n", "v" }, "<C-->", ":lua vim.g.neovide_scale_factor = vim.g.neovide_scale_factor - 0.1<CR>")
    vim.keymap.set({ "n", "v" }, "<C-0>", ":lua vim.g.neovide_scale_factor = 1<CR>")
end
```

**Comment by max2000warlord** [comment]
> This works flawlessly and is the simplest solution, hope it helps!
> 
> ```lua
> if vim.g.neovide then
>     map.set({ "n", "v" }, "<C-+>", ":lua vim.g.neovide_scale_factor = vim.g.neovide_scale_factor + 0.1<CR>")
>     map.set({ "n", "v" }, "<C-->", ":lua vim.g.neovide_scale_factor = vim.g.neovide_scale_factor - 0.1<CR>")
>     map.set({ "n" , "v" }, "<C-0>", ":lua vim.g.neovide_scale_factor = 1<CR>")
> end
> ```

What file is that in?

**Reply by fredizzimo** [reply]
It shouldn't matter, but if you are unsure then `init.lua` is fine.

But note that this solution can cause this crash
* https://github.com/neovide/neovide/issues/2221 

The [other solution above](https://github.com/neovide/neovide/discussions/2301#discussioncomment-8223201) is fine though, since it does not let the scale go down to 0. And you can change 0.5 to something smaller if you to allow it.

**Reply by palsmo** [reply]
The error is handled silently, no problem.

**Reply by palsmo** [reply]
@max2000warlord code was updated a tiny bit, place it in your `init.lua` file or with `require` link a different file containing your keybindings :+1:

**Comment by axelhj** [comment]
Thanks for this! Just in case it saves someone else 7 seconds on google; I modified warlords example to change the scaling with the scrollwheel just like Nvy or VSCode.

```
  vim.keymap.set({ "n", "v" }, "<C-ScrollWheelUp>", ":lua vim.g.neovide_scale_factor = vim.g.neovide_scale_factor + 0.1<CR>")
  vim.keymap.set({ "n", "v" }, "<C-ScrollWheelDown>", ":lua vim.g.neovide_scale_factor = vim.g.neovide_scale_factor - 0.1<CR>")
```
